lib/iris/util.py

```diff
@@ -1074,29 +1074,12 @@ def format_array(arr):
             separator=", ",
             edge_items=edge_items,
             summary_insert=summary_insert,
-            legacy=legacy,
+            legacy=format_options["legacy"],
```

Comment:
```suggestion
            legacy=False,
```

Numpy is programmed with handling for `False`, and this would avoid us importing the private `_format_options`. I can be convinced otherwise!

Reply:
Note all our tests still pass when this change is included. Is that evidence enough?

Reply:
@trexfeathers We don't want `legacy=False` here.

You can see that from the different implementations of `numpy.core.arrayprint._formatArray` for 1.21.5 vs 1.22.0

We can use the safe route of setting the `legacy` option through the public API of `numpy.printoptions` context manager and rely on it correctly translating it to the correct internal value within the `numpy.core.arrayprint._format_options` dictionary for the `"legacy"` key.

Reply:
> Note all our tests still pass when this change is included. Is that evidence enough?

I think it is. What else is there to do?

We're not in the business of testing a matrix of `numpy` versions against different `python` and `iris` versions.

... but I do know that this PRs changes work for `numpy` 1.21.5 **and** 1.22.0. That should be sufficient, right?

Reply:
But since it's within the context manager already, providing the key doesn't make any difference?

Reply:
>I think it is. What else is there to do?

I was referring in this case to the tests still passing even when using `legacy=False`.

Reply:
I think I can see the point of doing this if the new code would remain compatible with **_older_** versions of numpy.
**But surely we have already burned that bridge by adopting the context-manager form ??**
It seems a lot to assume that this will save us from _future_ changes.

TBH I'd much rather we had a stable, supported mechanism for this printed output.
It's caused so much work over the years, and it's absolutely bound to change again.
But equally there is probably a lot of clever stuff in the formatter, and I think even a minimal replacement would require a lot of work.

Reply:
> I was referring in this case to the tests still passing even when using `legacy=False`.

What does this matter? True, this is an interesting curiosity, but setting `legacy=False` is not the correct thing to do here. It's simply not future proof, particularly if we come to change `iris.util.format_array` and pin it to a future legacy style, such as `1.21`, as suggested as an option in `numpy` 1.22.0

The behaviour that we see from `format_array` is directly dependent on the data that is being printed. Setting `legacy=False` does not guarantee that the data of a future `iris` test will be truly 1.13 legacy compliant... regardless of whether all the `iris` tests are currently passing at the moment with `legacy=False`.

The doc-string for the `legacy` kwarg on `set_printoptions` is:
```
    legacy : string or `False`, optional
        If set to the string `'1.13'` enables 1.13 legacy printing mode. This
        approximates numpy 1.13 print output by including a space in the sign
        position of floats and different behavior for 0d arrays. If set to
        `False`, disables legacy mode. Unrecognized strings will be ignored
        with a warning for forward compatibility.
```
i.e., `legacy=False` means **disables legacy mode**. Note that, there is special treatment of the sign for floats and 0d arrays for `legacy="1.13"`. This is **NOT** the case for `legacy=False`, as is plain to see in the `numpy.core.arrayprint` code base for both `numpy` 1.21.5 and 1.22.0

Also, calling `numpy.core.arrayprint._formatArray` with `legacy=False` within the `numpy.printoptions` context-manager scope is just plain confusing to the reader, simply because the `legacy="1.13"` kwarg has just been passed to context-manager.

Note that I've attempted to introduce the minimum changes required here to continue successful `numpy` legacy 1.13 array printing for the `iris` supported versions of `numpy` (>=1.19) using `py38`. I'm not condoning the use of private `numpy` functionality, and I'm not going to attempt to re-write our approach to array printing, I'm simply attempting to unblock `iris` for the 3.2.0 release, given the diminishing time-scale.

If you don't want to merge this PR, then I see the following options:
1. the `iris` 3.2.0 can go ahead, but we can't test successfully against `numpy` +1.22.0, and `main` is also pinned to `numpy` <1.22.0 [:-1: from me]
2. invest more time and effort with a different strategy and PR for `iris` array printing of `numpy` arrays [personally, I'm not going to invest effort to do that at time]

Your call.

Reply:
Following discussion, I think this will do for now.

I think we shouldn't try to come up with a better way for this now, but **_should_** review it after the release, as outlined in [this recent comment on the longstanding issue about it](https://github.com/SciTools/iris/issues/3048#issuecomment-1009744727)